jetty-http2/http2-client/src/main/java/org/eclipse/jetty/http2/client/HTTP2ClientConnectionFactory.java

```diff
@@ -54,27 +56,30 @@ public Connection newConnection(EndPoint endPoint, Map<String, Object> context)
         Scheduler scheduler = client.getScheduler();
         Session.Listener listener = (Session.Listener)context.get(SESSION_LISTENER_CONTEXT_KEY);
         @SuppressWarnings("unchecked")
-        Promise<Session> promise = (Promise<Session>)context.get(SESSION_PROMISE_CONTEXT_KEY);
+        Promise<Session> sessionPromise = (Promise<Session>)context.get(SESSION_PROMISE_CONTEXT_KEY);
 
-        Generator generator = new Generator(byteBufferPool, client.getMaxDynamicTableSize(), client.getMaxHeaderBlockFragment());
+        Generator generator = new Generator(byteBufferPool, client.isUseOutputDirectByteBuffers(), client.getMaxHeaderBlockFragment());
         FlowControlStrategy flowControl = client.getFlowControlStrategyFactory().newFlowControlStrategy();
-        HTTP2ClientSession session = new HTTP2ClientSession(scheduler, endPoint, generator, listener, flowControl);
+
+        Parser parser = new Parser(byteBufferPool, client.getMaxDecoderTableCapacity(), client.getMaxResponseHeadersSize());
```

Comment:
I think technically the `getMaxDecoderTableCapacity()` should only be set on the decoder when the settings frame is about to be sent. Otherwise the application could specify in `SETTINGS` that they do not want to use the dynamic table and it will be ignored.

So this is too early.


I think we are already doing this in `HTTP2Session.configure()` so you should pass a 0 value for the `maxDecoderTableCapacity` to the `Parser` constructor.